.github/workflows/main.yml

```diff
@@ -0,0 +1,42 @@
+# © 2019-2023 Serokell <hi@serokell.io>
+# © 2019-2023 Lars Jellema <lars.jellema@gmail.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+env:
+  CDN_DISTRIBUTION_ID: E13UN1J1JFIWUZ
+  CDN_BUCKET: s3://nixfmt.serokell.io
+
+jobs:
+  check:
+    runs-on: [self-hosted, nix]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: reuse lint
+        run: nix shell .#packages.x86_64-linux.reuse -c reuse lint
+
+      - name: build
+        run: nix build .#nixfmt-static
```

Comment:
Probably should check that `.#nixfmt-webdemo` and `.#nixfmt-deriver` build as well though I'm not sure what the latter is. Maybe `.#awscli` too to be sure that the deploy action will work after nixpkgs upgrades. Put them all in separate checks if possible.

Reply:
Added checks for `nixfmt-webdemo` and `awscli`. `nixfmt-deriver` is a nix expression generated by `cabal2nix`, it was pushed into `cachix`, which I guess no one uses. I also added `hlint` and `stylish-haske`ll checks, but decided not to apply the changes to avoid large rebases in other pull requests.